src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -43,7 +43,7 @@ internal class RestClientBuilder
         public RestClientBuilder(IEnumerable<InputParameter> clientParameters, TypeFactory typeFactory)
         {
             _typeFactory = typeFactory;
-            _parameters = clientParameters.ToDictionary(p => p.Name, BuildConstructorParameter);
+            _parameters = clientParameters.DistinctBy(p => p.Name).ToDictionary(p => p.Name, BuildConstructorParameter);
         }
```

Comment:
For versioned service, if one has query apiversion and the other has path apiversion, there will be two apiversion parameters.